src/shogun/kernel/GaussianKernel.cpp

```diff
@@ -60,7 +67,15 @@ std::shared_ptr<SGObject >GaussianKernel::shallow_copy() const
 	// this assert is to avoid any subclass of GaussianKernel accidentally called
 	// with the implement here
 	ASSERT(typeid(*this) == typeid(GaussianKernel))
-	auto ker = std::make_shared<GaussianKernel>(cache_size, get_width());
+	std::shared_ptr<GaussianKernel> ker;
```

Comment:
I think you could just remove this method (is it even used?)

Reply:
I'll have a look, it did look very odd to me, as this is essentially done by the copy constructor..

Reply:
I dont think we ever really used those things, they were introduced as some form of hackery in some early gsoc.

Reply:
OK, I'll delete this function

Reply:
actually there is a handful of places that has this so I'll start an issue and can be done in different PR